dataduct/database/database.py

```diff
@@ -105,9 +105,9 @@ def has_cycles(self, relation=None, visited=None):
             if relation.full_name in visited:
                 return True
             # Make a copy for immutability
-            visited = deepcopy(visited)
-            visited.append(relation.full_name)
-            if self.has_cycles(relation, visited):
+            visited_copy = deepcopy(visited)
```

Comment:
The name `visited` is taken by a parameter already. Its copy is overwriting itself.

This fixes the cyclic dependency bug I asked you about yesterday.